Mathlib/RingTheory/Coalgebra/Convolution.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2025 Yaël Dillies, Michał Mrugała, Yunzhou Xie. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Michał Mrugała, Yunzhou Xie
+-/
+import Mathlib.LinearAlgebra.TensorProduct.Tower
+import Mathlib.RingTheory.Coalgebra.Basic
+
+/-!
+# Convolution product on linear maps from a coalgebra to an algebra
+
+This file constructs the ring structure on linear maps `C → A` where `C` is a coalgebra and `A` an
+algebra, where multiplication is given by
+```
+         .
+        / \
+f * g = f g
+        \ /
+         .
```

Comment:
I can't really parse this diagram.

Reply:
How would you render it?

Reply:
I personally would just write `(f * g)(x) = ∑ f x₍₁₎ * g x₍₂₎` in sweedler notation. It would be nice with a diagram though and perhaps you just need to explain what
```
 .
/ \
```
means (which I don't understand)